main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidatorTest.java

```diff
@@ -117,9 +107,51 @@ public void allNamesProvidedAndDifferentFromRouteDescShouldNotGenerateNotice() {
   @Test
   public void equalRouteShortNameRouteLongNameAndRouteDescShouldGenerateTwoNotices() {
     assertThat(validateRoute(createRoute("duplicate", "duplicate", "duplicate")))
-        .contains(new RouteShortAndLongNameEqualNotice("r1", 1, "duplicate", "duplicate"));
+        .contains(new RouteLongNameContainsShortNameNotice("r1", 1, "duplicate", "duplicate"));
     assertThat(validateRoute(createRoute("duplicate", "duplicate", "duplicate")))
         .contains(
             new SameNameAndDescriptionForRouteNotice(1, "r1", "duplicate", "route_short_name"));
   }
+
+  // test that notice is thrown if route_long_name begins with route_short_name followed by " ",
+  // "-", or "("
+  @Test
+  public void routeShortAndLongNameEqual() {
+    assertThat(validateRoute(createRoute("S", "S", null)))
+        .containsExactly(new RouteLongNameContainsShortNameNotice("r1", 1, "S", "S"));
+    // Compare case-insensitive.
+    assertThat(validateRoute(createRoute("SA", "Sa", null)))
+        .containsExactly(new RouteLongNameContainsShortNameNotice("r1", 1, "SA", "Sa"));
+
+    assertThat(validateRoute(createRoute("S", "Long", null))).isEmpty();
+  }
+
+  @Test
+  public void routeLongNameContainsRouteShortNameShouldGenerateNotice() {
+    List<String> badValues =
+        Arrays.asList(
+            new String[] {
+              // shortName, longName
+              "L1 Long Name",
+              "L1-Long Name",
+              "L1- Long Name",
+              "L1 - Long Name",
+              "L1)Long Name",
+              "L1) Long Name",
+              "L1 ) Long Name",
+              "L1(Long Name)",
+              "L1 (Long Name)",
+              "L1( Long Name)",
+            });
```

Comment:
You don't need the new `String[] { }` call here, as Arrays.asList(...) accepts a var-arg parameter list.